Test cases for backend fallback kernels

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30370 [wip] Use codegen'ed unboxing wrappers
* #29934 Boxed variable dispatch
* #29682 backend fallback test
* #29681 better boxing
* #29337 Hide the OperatorKernel* argument from the stack based kernel API
* #29330 OperatorHandle::callBoxed/callUnboxed
* **#29214 Test cases for backend fallback kernels**
* #29201 Add a OperatorHandle argument to boxed kernels
* #30340 Make Dispatcher::backendFallbackKernels_ an array
* #30333 [pytorch][PR] Remove LeftRight from OperatorEntry and DispatchTable.
* #30332 [pytorch][PR] Convert KernelTable to a flat-indexed array rather than a hashtable.

-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

smessmer committed:
Test cases for backend fallback kernels

-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

ezyang reviewed (approved):
These tests are fine but they are not the same as the old boxed fallback tests, which showed you how to use backend fallback to do something more similar to what you /actually/ want to do (fallthrough to the underlying kernel, and do some other stuff along the way.)

ezyang commented:
(So, I'm not blocking landing this PR, but I don't think @mcarilli will really be satisfied by these tests.)

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer commented:
@ezyang @mcarilli added those ones in https://github.com/pytorch/pytorch/pull/29682

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]

smessmer committed:
Update on "Test cases for backend fallback kernels"


-

Differential Revision: [D18329308](https://our.internmc.facebook.com/intern/diff/D18329308/)

[ghstack-poisoned]
